File: spring/boot/microservices/with_cloud/with_eventmessaging/with_kafka/productms2/src/main/java/org/example/productms/entities/Product.java
package org.example.productms.entities;

import lombok.*;

import javax.persistence.*;
import java.util.Objects;

@Data
@NoArgsConstructor
@EqualsAndHashCode(of = "id")
@NamedQuery(name="Product.findByNameInAscending",query="from Product where name=:name order by price")
@Entity
public class Product {
    @GeneratedValue
    @Id
    private Long id;
    @NonNull
    @Column(unique = true)
    private String name;
    @NonNull
    private double price;
    private int units;
}

File: frontend/angular/second-app/src/app/product.ts
export class Product {
  private _price: number = -1;
  private _name: string = "";
  constructor(name: string, price: number) {
    this._price = price;
    this._name = name;
  }

  get price(): number {
    return this._price;
  }
  set price(_price: number) {
    this._price = _price;
  }

  get name(): string {
    return this._name;
  }
  set name(name: string) {
    this._name = name;
  }

}
